kbl2vec/src/main/java/com/foursoft/harness/kbl2vec/transform/placements/OnWayPlacementTransformer.java

```diff
@@ -0,0 +1,60 @@
+/*-
+ * ========================LICENSE_START=================================
+ * KBL to VEC Converter
+ * %%
+ * Copyright (C) 2025 4Soft GmbH
+ * %%
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ * =========================LICENSE_END==================================
+ */
+package com.foursoft.harness.kbl2vec.transform.placements;
+
+import com.foursoft.harness.kbl.v25.KblProtectionArea;
+import com.foursoft.harness.kbl2vec.core.Query;
+import com.foursoft.harness.kbl2vec.core.TransformationContext;
+import com.foursoft.harness.kbl2vec.core.TransformationResult;
+import com.foursoft.harness.kbl2vec.core.Transformer;
+import com.foursoft.harness.vec.v2x.VecOnWayPlacement;
+import com.foursoft.harness.vec.v2x.VecPlaceableElementRole;
+import com.foursoft.harness.vec.v2x.VecSegmentLocation;
+
+public class OnWayPlacementTransformer implements Transformer<KblProtectionArea, VecOnWayPlacement> {
+
+    @Override
+    public TransformationResult<VecOnWayPlacement> transform(final TransformationContext context,
+                                                             final KblProtectionArea source) {
+        final VecOnWayPlacement destination = new VecOnWayPlacement();
+        destination.setIdentification(source.getAssociatedProtection().getId());
+
+        return TransformationResult.from(destination)
+                .withDownstream(KblProtectionArea.class, VecSegmentLocation.class, Query.of(source),
+                                this::setLocation)
+                .withLinker(Query.of(source::getAssociatedProtection), VecPlaceableElementRole.class,
+                            VecOnWayPlacement::getPlacedElement)
+                .build();
+    }
+
+    final void setLocation(final VecOnWayPlacement destination, final VecSegmentLocation location) {
+        if (location.getIdentification().equals(Constants.START)) {
```

Comment:
Comment on robust programming: Constants and Enums first! Make it:

```java
Constants.START.equals(location.getIdentification()) 
```
That way, you will never get a NullPointException, because const is always non null. The `getIdentifaction()` might return null (not in the current case, because you set it by yourself in a differnt class, but you never know.